src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java

```diff
@@ -0,0 +1,180 @@
+            } else {
+                log.warn("Player not found");
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public Teams saveTeams(Teams team) {
+        if (teams.size() >= teamLimit) {
+            log.warn("Team limit exceeded");
+            return null;
+        }
+        log.debug("Saving team {}", team);
+        team.setId(nextTeam++);
+        teams.add(team);
+
+        return team;
+    }
+
+    @Override
+    public Teams findTeams(int id) {
+        for (Teams team : teams) {
+            if (team.getId() == id) {
+                return team;
+            } else {
+                log.warn("Team not found");
+                return null;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Matches saveMatches(Matches match) {
+        log.debug("Saving match {}", match);
+        match.setIdMatch(nextMatches++);
+        matches.add(match);
+        return match;
+    }
+
+    public Matches findMatches(int id) {
+        for (Matches match : matches) {
+            if (match.getIdMatch() == id) {
+                return match;
+            } else {
+                log.warn("Match not found");
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<Matches> getMatchesTeam(String nameTeam) {
+        List<Matches> matchesTeam = new ArrayList<>();
+
+        for (Matches match : matches) {
+            if (match.getNameTeam1().equals(nameTeam)) {
+                matchesTeam.add(match);
+            } else if (match.getNameTeam2().equals(nameTeam)) {
+                matchesTeam.add(match);
+            }
+        }
+
+        return matchesTeam;
+    }
+
+    @Override
+    public Players updatePlayer(int id, Players player) {
+        Players existingPlayer = findPlayers(id);
+        if (existingPlayer == null) {
+            log.warn("Player with id {} not found", id);
+            return null;
+        } else if (findTeams(player.getTeamId()) == null) {
+            log.warn("Team with id {} not found", player.getTeamId());
+            return null;
+        }
+
+        existingPlayer.setName(player.getName());
+        existingPlayer.setTeamId(player.getTeamId());
+
+        return existingPlayer;
+    }
+
+    @Override
+    public Teams updateTeam(int id, Teams team) {
+        Teams existingTeam = findTeams(id);
+        if (existingTeam == null) {
+            log.warn("Team with id {} not found", id);
+            return team;
+        }
+
+        existingTeam.setName(team.getName());
+
+        return existingTeam;
+    }
+
+
+    @Override
+    public List<Players> getPlayersTeam(int teamId) {
+        return players.stream().filter(player -> player.getTeamId() == teamId).collect(Collectors.toList());
+    }
+
+    @Override
+    public Teams deletePlayersTeam(int teamId) {
+        for (Teams team : teams) {
+            if (team.getId() == teamId) {
+                team.getPlayers().clear();
```

Comment:
Before clearing this you will also to delete the value of each Player so they dont have a team